go.mod

```diff
@@ -1,10 +1,11 @@
 module github.com/operator-framework/rukpak
 
-go 1.21
+go 1.22
```

Comment:
Is there a technical reason for this to bump golang also?  I think we'd prefer to keep potentially disruptive dep bumps (for e.g. k8s/controller-runtime/kubebuilder) discrete from feature/bug PRs in general.

Reply:
The version of kapp we are pulling in as a dependency uses go 1.22

Reply:
Ah.  I see with 1.21++ it [became mandatory](https://go.dev/ref/mod#go-mod-file-go):
```
The go line must be greater than or equal to the go line of all dependencies.
```

Boo.

Reply:
Yeah, there are more preflight checks we could pull in but it resulted in k8s version bumps to v0.30.z and so I refrained from going further than this PR currently does.

It looks like the tilt stuff is failing on this go bump as well...

Reply:
Rebased on main which already bumped go and k8s deps.